marlowe/src/Language/Marlowe/Client.hs

```diff
@@ -548,14 +548,16 @@ mkMarloweStateMachineTransition params SM.State{ SM.stateData=MarloweData{..}, S
                     marloweContract = txOutContract,
                     marloweState = txOutState }
 
-            let (outputsConstraints, finalBalance) = case txOutContract of
-                    Close -> (payoutConstraints txOutPayments, P.zero)
-                    _ -> let
-                        outputsConstraints = payoutConstraints txOutPayments
-                        totalIncome = P.foldMap collectDeposits inputs
-                        totalPayouts = P.foldMap (\(Payment _ _ v) -> v) txOutPayments
-                        finalBalance = totalIncome P.- totalPayouts
-                        in (outputsConstraints, finalBalance)
+            let (outputsConstraints, finalBalance) = let
+                    payoutsByParty = AssocMap.toList $ P.foldMap payoutByParty txOutPayments
+                    in case txOutContract of
+                        Close -> (payoutConstraints payoutsByParty, P.zero)
+                        _ -> let
+                            outputsConstraints = payoutConstraints payoutsByParty
+                            totalIncome = P.foldMap collectDeposits inputs
+                            totalPayouts = P.foldMap snd payoutsByParty
+                            finalBalance = inputBalance P.+ totalIncome P.- totalPayouts
```

Comment:
Ah, so it was probably from the change to include transfers in the payments, didn't see that one